inference-engine/src/mkldnn_plugin/emitters/jit_load_store_emitters.cpp

```diff
@@ -156,104 +156,114 @@ void jit_load_emitter::load_bytes(const Vmm &vmm, const Xbyak::Reg64 &reg, int o
-                    h->uni_vpinsrw(xmm, xmm, addr(start_bytes + 4), 2);
-                    h->uni_vpinsrb(xmm, xmm, addr(start_bytes + 6), 6);
-                    break;
-                case 8: break;
-                case 9: h->uni_vpinsrb(xmm, xmm, addr(start_bytes + 8), 8); break;
-                case 10: h->uni_vpinsrw(xmm, xmm, addr(start_bytes + 8), 4); break;
-                case 11:
-                    h->uni_vpinsrw(xmm, xmm, addr(start_bytes + 8), 4);
-                    h->uni_vpinsrb(xmm, xmm, addr(start_bytes + 10), 10);
-                    break;
-                case 12: h->pinsrd(xmm, addr(start_bytes + 8), 2); break;
-                case 13:
-                    h->pinsrd(xmm, addr(start_bytes + 8), 2);
-                    h->uni_vpinsrb(xmm, xmm, addr(start_bytes + 12), 12);
-                    break;
-                case 14:
-                    h->pinsrd(xmm, addr(start_bytes + 8), 2);
-                    h->uni_vpinsrw(xmm, xmm, addr(start_bytes + 12), 6);
-                    break;
-                case 15:
-                    h->pinsrd(xmm, addr(start_bytes + 8), 2);
-                    h->uni_vpinsrw(xmm, xmm, addr(start_bytes + 12), 6);
-                    h->uni_vpinsrb(xmm, xmm, addr(start_bytes + 14), 14);
-                    break;
-                case 16: break;
-                default:
-                    IE_THROW() << "Load emitter in " << name<< " has unexpected number of values to load in load_byte.";
-            }
+        if (has_xmm_block) {
+            h->vinsertf128(ymm, ymm, xmm, 1); // insert to upper bits of ymm
+            if (has_ymm_block)
+                h->vinsertf128(ymm, ymm, addr(32), 0); // insert to lower bits of ymm
+            else
+                h->vinsertf128(ymm, ymm, addr(0), 0); // insert to lower bits of ymm
+        }
 
-            if (has_xmm_block) {
-                h->vinsertf128(ymm, ymm, xmm, 1); // insert to upper bits of ymm
-                if (has_ymm_block)
-                    h->vinsertf128(ymm, ymm, addr(32), 0); // insert to lower bits of ymm
-                else
-                    h->vinsertf128(ymm, ymm, addr(0), 0); // insert to lower bits of ymm
-            }
+        if (has_ymm_block) {
+            h->vinsertf64x4(zmm, zmm, ymm, 1); // insert to upper bits of zmm
+            h->vinsertf64x4(zmm, zmm, addr(0), 0); // insert to lower bits of zmm
+        }
+    };
 
-            if (has_ymm_block) {
-                h->vinsertf64x4(zmm, zmm, ymm, 1); // insert to upper bits of zmm
-                h->vinsertf64x4(zmm, zmm, addr(0), 0); // insert to lower bits of zmm
+    switch (load_size) {
+        case 64:
+            h->uni_vmovdqu(zmm, addr(0));
+            break;
+        case 32:
+            h->uni_vmovdqu(ymm, addr(0));
+            break;
+        case 16:
+            h->uni_vmovdqu(xmm, addr(0));
+            break;
+        default: {
+            if (mayiuse(cpu::x64::avx512_core)) {
```

Comment:
Just wondering is this branch valid for cases Vmm != Zmm?

Reply:
yes, Ymm and Xmm is also covered. mayiuse(cpu::x64::avx512_core) branch covers [0, 64] numbers 8bit data loading on avx512_core target. load_byte_base() cover [0, 64] numbers 8bit data loading on target before avx512_core is enabled. If tails length is [0, 16], only sse41 and before ISA is used. If tails length is [0, 32], only avx2 and before(like sse41) ISA is used. If tails length is [0, 64], only avx512_common and before ISA is used. Thank you!

Reply:
and there is [checks ](https://github.com/openvinotoolkit/openvino/pull/6771/files#diff-b7601237911270beac852b876494e1c3f3d37206f9341910528e5bab8963ffecR145-R148) that if Xmm(sse41) version, the load length should not be larger than 16(126 bit). if Ymm(avx2) version, the load length should not be larger than 32(256 bit).